### ISSUE TITLE
Database credentials visible in logs

### ISSUE THREAD

**Issue by peterkingsbury** [heroiclabs/nakama#461]
## Description
Database credentials (username+password) visible in logs on startup at numerous points during startup.

This is less of a concern when Nakama is running isolated from other services, but when logs are shipped to external sources (e.g. Elastic or others) anyone with access to those external tools has access to database credentials.

### Assumptions
A database (I used PostgreSQL, but CockroachDb would be fine) with the setup indicated in the `.env` file.

### Steps
1. Start with a docker-compose file similar to [Install Docker Quickstart](https://heroiclabs.com/docs/install-docker-quickstart/).
2. Modify the docker-compose.yml file to obtain credentials from the environment:
```
...
    env_file:
      - ./.env
    entrypoint:
      - "/bin/sh"
      - "-ecx"
      - >
        /nakama/nakama migrate up --database.address ${POSTGRES_USER}:${POSTGRES_PASSWORD}@${POSTGRES_HOST}:${POSTGRES_PORT}/${POSTGRES_DB} &&
        exec /nakama/nakama --name nakama1 --database.address ${POSTGRES_USER}:${POSTGRES_PASSWORD}@${POSTGRES_HOST}:${POSTGRES_PORT}/${POSTGRES_DB} --console.password=${CONSOLE_PASSWORD}
...
```
3. Add an `.env` file with the following:
```
POSTGRES_USER=MYDBUSERNAME
POSTGRES_PASSWORD=MYSUPERSECRETPASSWORD
POSTGRES_HOST=MYDBHOST
POSTGRES_PORT=MYDBPORT
POSTGRES_DB=MYDB
CONSOLE_PASSWORD=MYCONSOLEPASSWORD
```
4. Start Nakama with `docker-compose up -d`.
5. Observe log output using `docker logs -f nakama`.
6. Compare log output to XHR `GET /v2/console/config?` in web console.

## Expected Result
Database credentials are obscured before being delivered to stdout/stderr (e.g. `MYDBUSERNAME:********@MYDBHOST:MYDBPORT/MYDB` instead of `MYDBUSERNAME:MYSUPERSECRETPASSWORD@MYDBHOST:MYDBPORT/MYDB`).

Comparing this with Config section in the web console, the console scrubs credentials (database password, console password) before sending to a UI client.

## Actual Result
Credentials clearly indicated as both part of the command-execution being sent to the container (unsure whether this can be fixed) as well as the Nakama log output itself:
```
+ /nakama/nakama migrate up --database.address MYDBUSERNAME:MYSUPERSECRETPASSWORD@MYDBHOST:MYDBPORT/MYDB
{"level":"info","ts":"2020-09-19T18:44:14.140Z","caller":"migrate/migrate.go:139","msg":"Database connection","dsn":"MYDBUSERNAME:MYSUPERSECRETPASSWORD@MYDBHOST:MYDBPORT/MYDB"}
{"level":"info","ts":"2020-09-19T18:44:14.153Z","caller":"migrate/migrate.go:154","msg":"Database information","version":"PostgreSQL 12.4 on x86_64-pc-linux-musl, compiled by gcc (Alpine 9.3.0) 9.3.0, 64-bit"}
{"level":"info","ts":"2020-09-19T18:44:14.154Z","caller":"migrate/migrate.go:158","msg":"Using existing database","name":"MYDB"}
{"level":"info","ts":"2020-09-19T18:44:14.165Z","caller":"migrate/migrate.go:192","msg":"Successfully applied migration","count":0}
+ exec /nakama/nakama --name nakama1 --database.address MYDBUSERNAME:MYSUPERSECRETPASSWORD@MYDBHOST:MYDBPORT/MYDB --console.password=MYCONSOLEPASSWORD
{"level":"warn","ts":"2020-09-19T18:44:14.241Z","caller":"server/config.go:242","msg":"WARNING: insecure default parameter value, change this for production!","param":"console.username"}
{"level":"warn","ts":"2020-09-19T18:44:14.241Z","caller":"server/config.go:250","msg":"WARNING: insecure default parameter value, change this for production!","param":"console.signing_key"}
{"level":"warn","ts":"2020-09-19T18:44:14.241Z","caller":"server/config.go:254","msg":"WARNING: insecure default parameter value, change this for production!","param":"socket.server_key"}
{"level":"warn","ts":"2020-09-19T18:44:14.241Z","caller":"server/config.go:258","msg":"WARNING: insecure default parameter value, change this for production!","param":"session.encryption_key"}
{"level":"warn","ts":"2020-09-19T18:44:14.241Z","caller":"server/config.go:262","msg":"WARNING: insecure default parameter value, change this for production!","param":"runtime.http_key"}
{"level":"info","ts":"2020-09-19T18:44:14.241Z","caller":"main.go:104","msg":"Nakama starting"}
{"level":"info","ts":"2020-09-19T18:44:14.241Z","caller":"main.go:105","msg":"Node","name":"nakama1","version":"2.13.0+ec8f5d70","runtime":"go1.15","cpu":2,"proc":2}
{"level":"info","ts":"2020-09-19T18:44:14.241Z","caller":"main.go:106","msg":"Data directory","path":"/nakama/data"}
{"level":"info","ts":"2020-09-19T18:44:14.241Z","caller":"main.go:107","msg":"Database connections","dsns":["MYDBUSERNAME:MYSUPERSECRETPASSWORD@MYDBHOST:MYDBPORT/MYDB"]}
```

## Context
- Server only.

## Your Environment

Nakama Docker image is `heroiclabs/nakama:2.13.0`.
```
$ docker exec -it nakama ./nakama --version
2.13.0+ec8f5d70

$ docker -v
Docker version 19.03.13, build 4484c46d9d

$ docker-compose -v
docker-compose version 1.25.0, build unknown
```
Not running in Swarm mode.

Edit: Added example `.env` file. Added assumptions.

**Comment by peterkingsbury** [comment]
PR here: https://github.com/heroiclabs/nakama/pull/467

**Comment by zyro** [comment]
Fixed in f0fe70147c1545151aa05b07b8bb586ae3ab14a9 and will be part of the next release.
